packages/cosmos/src/client.rs

```diff
@@ -147,6 +148,131 @@ impl<Res> PerformQueryWrapper<Res> {
 }
 
 impl Cosmos {
+    async fn get_and_update_simulation_sequence(
+        &self,
+        address: Address,
+    ) -> Result<BaseAccount, Error> {
+        let mut guard = self.pool.get().await?;
+        let cosmos = guard.get_inner_mut();
+        let sequences = cosmos.simulate_sequences.read().await;
+        let sequence = sequences.get(&address);
+        let mut base_account = self.get_base_account(address).await?;
+        if let Some(SequenceInformation {
+            sequence,
+            timestamp,
+        }) = sequence
+        {
+            let diff = Instant::now().duration_since(*timestamp);
+            if diff.as_secs() <= 30 {
+                let max_sequence = std::cmp::max(sequence, &base_account.sequence);
+                if max_sequence != sequence {
+                    let sequence_info = SequenceInformation {
+                        sequence: *max_sequence,
+                        timestamp: Instant::now(),
+                    };
+                    {
+                        let mut seq_info = cosmos.simulate_sequences.write().await;
+                        let seq_info = seq_info
+                            .entry(address)
+                            .or_insert_with(|| sequence_info.clone());
+                        *seq_info = sequence_info;
+                    }
+                }
+                base_account.sequence = *max_sequence;
+                return Ok(base_account);
+            }
+        }
+        let mut seq_info = cosmos.simulate_sequences.write().await;
+        let sequence_info = SequenceInformation {
+            sequence: base_account.sequence,
+            timestamp: Instant::now(),
+        };
+        let seq_info = seq_info
+            .entry(address)
+            .or_insert_with(|| sequence_info.clone());
+        *seq_info = sequence_info;
+        Ok(base_account)
+    }
+
+    async fn update_broadcast_sequence(
+        &self,
+        address: Address,
+        tx: &Tx,
+        hash: &str,
+    ) -> Result<(), Error> {
+        let mut guard = self.pool.get().await?;
+        let cosmos = guard.get_inner_mut();
+        let auth_info = &tx.auth_info;
+        if let Some(auth_info) = auth_info {
+            let sequence = &auth_info
+                .signer_infos
+                .iter()
+                .map(|item| item.sequence)
```

Comment:
This only works right now since we have a limitation of only allowing a single signer per transaction. Probably worth at least a comment about this.

Reply:
Added a comment.